rocAL/rocAL/include/loaders/image/image_read_and_decode.h

```diff
@@ -72,13 +72,16 @@ class ImageReadAndDecode
             std::vector<uint32_t> &actual_height,
             RocalColorFormat output_color_format,
             bool decoder_keep_original=false);
-
+    void feed_external_input(std::vector<std::string> input_images, std::vector<int> labels, std::vector<unsigned char *> input_buffer,
+                             std::vector<unsigned> roi_width, std::vector<unsigned> roi_height,
+                             unsigned int max_width, unsigned int max_height, FileMode mode, bool eos) ;
     //! returns timing info or other status information
     Timing timing();
 
-private:
+protected:
     std::vector<std::shared_ptr<Decoder>> _decoder;
     std::shared_ptr<Reader> _reader;
+    StorageType _reader_type;
```

Comment:
Rename it as _storage_type since reader type is inferred from the storage type and is not directly a reader type